misk/src/main/kotlin/misk/web/actions/WebActionFactory.kt

```diff
@@ -130,8 +132,20 @@ internal class WebActionFactory @Inject constructor(
    * Returns null if neither of the input action's content types were protobuf.
    */
   private fun transformActionIntoJson(action: Action): Action? {
-    var jsonVariant = action
+    // Map gRPC actions to the equivalent JSON actions. This changes the dispatch mechanism from
+    // gRPC to POST which prevents gRPC framing from being used.
+    if (action.dispatchMechanism == DispatchMechanism.GRPC && action.parameters.size == 1) {
```

Comment:
what does the parameter.size == 1 mean?

Reply:
gRPC streams have two parameters (inbound and outbound streams)

Reply:
k. a think a code comment would be helpful here